diff --color -Naur i3-4.17.1/i3bar/src/outputs.c i3-4.17.1-1/i3bar/src/outputs.c
--- i3-4.17.1/i3bar/src/outputs.c	2019-08-30 23:21:24.000000000 +0200
+++ i3-4.17.1-1/i3bar/src/outputs.c	2020-09-14 04:04:25.396729780 +0200
@@ -196,8 +196,8 @@
         bool handle_output = false;
         for (int c = 0; c < config.num_outputs; c++) {
             if (strcasecmp(params->outputs_walk->name, config.outputs[c]) == 0 ||
-                (strcasecmp(config.outputs[c], "primary") == 0 &&
-                 params->outputs_walk->primary)) {
+                (strcasecmp(config.outputs[c], "primary") == 0 && params->outputs_walk->primary) ||
+                (strcasecmp(config.outputs[c], "nonprimary") == 0 && !params->outputs_walk->primary)) {
                 handle_output = true;
                 break;
             }
